js/src/evaluation/llm_evaluator.ts

```diff
@@ -0,0 +1,248 @@
+import { initChatModel } from "langchain/chat_models/universal";
+import { ChatPromptTemplate } from "@langchain/core/prompts";
+import * as uuid from "uuid";
+import { EvaluationResult, EvaluationResults, RunEvaluator } from "./evaluator.js";
+import type { Run, Example } from "../schemas.js";
+export class CategoricalScoreConfig {
+  key: string;
+  choices: string[];
+  description: string;
+  includeExplanation: boolean;
```

Comment:
Not a fan of boolean flags. Would prefer something like "reasoningKey" that if not provided is undefined and not included (if that's what we want for default behavior)